Update `fastlane` to version 2.203.0 to get `scan` improvements

mokagio opened the pull request:
Recent versions of Fastlane introduced changes in `scan` which might help with a particular issue we're seeing on our end, where UI tests succeeds on an automatic retry, but the CI result is still a failure. E.g.:
https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-iOS/27149/workflows/a49cf831-2fd8-4e49-b3d2-f4fe9550c204/jobs/64928

For some of the recent `scan` changes, see:

- https://github.com/fastlane/fastlane/releases/tag/2.201.0
- https://github.com/fastlane/fastlane/releases/tag/2.201.1
- https://github.com/fastlane/fastlane/releases/tag/2.201.2
- https://github.com/fastlane/fastlane/releases/tag/2.203.0

To test this we could re-run the build a bunch of times, but I think we're better off merging on green and seeing how this performs. Unless the scan changes somehow don't match with our setup and result in consistent failures, staying up to date with Fastlane is a good thing.

## Regression Notes
1. Potential unintended areas of impact: `scan` and other Fastlane automation
2. What I did to test those areas of impact: nothing really, just letting CI do it's job
3. What automated tests I added (or what prevented me from doing so): N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

mokagio committed:
Update `fastlane` to version 2.203.0 to get `scan` improvements

Recent versions of Fastlane introduced changes in `scan` which might
help with a particular issue we're seeing on our end, where UI tests
succeeds on an automatic retry, but the CI result is still a failure.
E.g.:
https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-iOS/27149/workflows/a49cf831-2fd8-4e49-b3d2-f4fe9550c204/jobs/64928

For some of the recent `scan` changes, see:

- https://github.com/fastlane/fastlane/releases/tag/2.201.0
- https://github.com/fastlane/fastlane/releases/tag/2.201.1
- https://github.com/fastlane/fastlane/releases/tag/2.201.2
- https://github.com/fastlane/fastlane/releases/tag/2.203.0

AliSoftware reviewed (approved):
I'm always 👍  for updating to latest versions of our tools!

(I agree that we're better off merging on green and see how it goes rather than forcing reruns of the Unit Tests until we get a run where it fails-and-retry just to test this)

mokagio committed:
Remove unused `fastlane-plugin-test_center`

The plugin became unused in 75f87334ee, but at the time I forgot to
remove its import from the `Pluginfile`.

I noticed this because, after upgrading to Fastlane 2.203.0, we saw an
error which is apparently due to a name-clash between the `trainer` gem
the plugin requires and the `trainer` Fastlane component.

See:

- Failing build on CircleCI https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-iOS/27160/workflows/4b4839c8-9d33-4507-afa2-57644d03bc67/jobs/64957
- Failing build on Buildkite https://buildkite.com/wordpress-mobile/wordpress-ios/builds/4716#f3a4719e-6496-4a1e-8cca-051d3b9f083d/664-2569
- Fastlane issue https://github.com/fastlane/fastlane/issues/19834

jkmassel commented:
<!-- REUSE_ID: jetpack-installable-build-link -->You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17841-eda10d4</code>. IPA is available <a href='https://appcenter-filemanagement-distrib3ede6f06e.azureedge.net/c1c183ff-3d5e-4890-8579-e3aae512ab50/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=ZM9tcbOSAzUCu2%2Fb1TFIuC0z6ZDLcr%2B1Ix2UhI4AXo8%3D&se=2022-02-01T14%3A06%3A49Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

jkmassel commented:
<!-- REUSE_ID: installable-build-link -->You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17841-eda10d4</code>. IPA is available <a href='https://appcenter-filemanagement-distrib2ede6f06e.azureedge.net/f83a248a-2e56-4578-ae0d-890a26713ff2/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=k1xzc%2FydGzOJeVkN15zapC712OnrtqDYxiVq0shZKTc%3D&se=2022-02-01T14%3A08%3A02Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.
